config/raw_config.go

```diff
@@ -48,6 +48,18 @@ func NewRawConfig(raw map[string]interface{}) (*RawConfig, error) {
 	return result, nil
 }
 
+// RawMap returns a copy of the RawConfig.Raw map.
+func (r *RawConfig) RawMap() map[string]interface{} {
+	r.lock.Lock()
+	defer r.lock.Unlock()
+
+	m := make(map[string]interface{})
+	for k, v := range r.Raw {
+		m[k] = v
+	}
+	return m
+}
```

Comment:
As annoying as it is, I would add a test for `RawMap` that ensures:

  1. It points to a different map

  1. The contents (key/values) are `reflect.DeepEqual` to the map itself